homeassistant/components/mqtt/__init__.py

```diff
@@ -132,10 +134,24 @@ def publish_service(call):
         """Handle MQTT publish service calls."""
         msg_topic = call.data.get(ATTR_TOPIC)
         payload = call.data.get(ATTR_PAYLOAD)
-        qos = call.data.get(ATTR_QOS, DEFAULT_QOS)
-        retain = call.data.get(ATTR_RETAIN, DEFAULT_RETAIN)
+        if payload is None:
+            try:
+                payload_template = call.data.get(ATTR_PAYLOAD_TEMPLATE)
+                if payload_template is None:
+                    _LOGGER.error(
+                        "You must set either '%s' or '%s' to use this service",
+                        ATTR_PAYLOAD, ATTR_PAYLOAD_TEMPLATE)
+                    return
+                payload = template.render(hass, payload_template)
+            except AttributeError as exc:
```

Comment:
If this is to catch a template error, you should catch `TemplateError`
```python
from homeassistant.exceptions import TemplateError
```
Make sure you return after the error occurred.